MapboxNavigation/JunctionView.swift

```diff
@@ -50,18 +51,30 @@ public class JunctionView: UIImageView {
                 imageRepository.imageWithURL(guidanceViewImageURL, cacheKey: guidanceView.cacheKey!) { [unowned self] (downloadedImage) in
                     
                     self.isCurrentlyVisible = true
-                    self.isHidden = !self.isCurrentlyVisible
                     
                     DispatchQueue.main.async {
+                        self.isHidden = !self.isCurrentlyVisible
                         self.image = downloadedImage
-                        self.show(true)
                     }
                 }
             }
         }
         
     }
     
+    public func updateDistance(for currentStepProgress: RouteStepProgress) {
+        let distanceTravelled = currentStepProgress.distanceTraveled
+        if distanceAlongStep > -1 {
+            // show the Junction View if we have progressed enough along the step
+            // hide the Junction View if it is still visible from a previous instruction but shouldn't be yet.
+            if distanceTravelled >= distanceAlongStep, isHidden == true  {
+                show()
+            } else if distanceTravelled < distanceAlongStep, isHidden == false {
+                hide()
+            }
```

Comment:
Trying to make the logic a little easier to follow:

```suggestion
            var shouldHide = distanceTravelled < distanceAlongStep
            if isHidden && !shouldHide {
                show()
            } else if !isHidden && shouldHide {
                hide()
            }
```

Reply:
This is definitely clearer. I'll update my code.